### PR TITLE
feat(core/dfn-panel): run in exported documents

### PR DESCRIPTION
**Pull request by sidvishnoi**
This is a very hacky way. Anyone with better ideas?

### CODE REVIEW

File: tests/spec/core/dfn-panel-spec.js [hunk 1]
```diff
@@ -165,4 +165,46 @@ describe("Core — dfnPanel", () => {
...
   });
+
+  it("works in exported document", async () => {
+    const rdoc = await makeRSDoc(ops);
+    const doc = await getExportedDoc(rdoc);
+
+    const dfnOne = doc.querySelector("dfn#dfn-one");
+    dfnOne.click();
+    const panel = doc.getElementById("dfn-panel");
+    expect(panel).toBeTruthy();
```
Comment (sidvishnoi) [inline tests/spec/core/dfn-panel-spec.js]: works locally :thinking:

**Comment by saschanaz** [comment]
#1668 is related. I think we should make the whole module to be injected as a `<script>` so that it can run in a same way both in raw and exported cases.

File: tests/spec/core/dfn-panel-spec.js (hunk 1 above)
Comment (saschanaz) [reply tests/spec/core/dfn-panel-spec.js]: That GitHub action is so cool

**Comment by saschanaz** [comment]
My thought: If we have any states that can/must be computed early, the code may expose them as a `<script type="application/json">` so that a module as a `<script>` can use it.

**Comment by sidvishnoi** [comment]
How about create a runtime profile (`profiles/w3c-static.js`)? We can inject it after `fixup.js` (as a `script[src]`). My idea is to export a `main()` as static part of a module and make base-runner run `main` instead of `run`.

**Comment by saschanaz** [comment]
I think creating a new profile can introduce breaking changes too easily. I vote to inject as a inline script.

**Comment by marcoscaceres** [comment]
If it helps, the W3C can host the panel code for us. Inlining is 🤢.

**Comment by saschanaz** [comment]
>If it helps, the W3C can host the panel code for us. Inlining is 🤢.

What if we have to change the structure and it breaks old specs? Inlining can be 🤢 but backward compatibility is 🤢🤢.

**Comment by sidvishnoi** [comment]
We've decided to go with inlining. But we need to figure out a strategy for that. Inlining like in this PR is :nauseated_face:

**Comment by marcoscaceres** [comment]
Difficult tradeoffs here... but seems inline is the way to go.

**Comment by sidvishnoi** [comment]
> My thought: If we have any states that can/must be computed early, the code may expose them as a `<script type="application/json">` so that a module as a `<script>` can use it.

@saschanaz can you expand on this approach? I'm not liking the way I've done in this PR, so I'm open to other ideas.

**Comment by sidvishnoi** [comment]
> The existing dfn-panel module inserts this new runtime module into the document

We would still need to serialize using `fn.toString()`, right?

**Comment by saschanaz** [comment]
>We would still need to serialize using fn.toString(), right?

No, we would get the runtime module as a string and then `<script>${module}</script>`.

File: src/core/dfn-panel.js [hunk 2]
```diff
@@ -67,15 +74,15 @@ function createPanel(dfn) {
   const href = `#${id}`;
   const links = document.querySelectorAll(`a[href="${href}"]`);
 
-  /** @type {HTMLElement} */
-  const panel = hyperHTML`
+  const tempNode = document.createElement("div");
+  tempNode.innerHTML = `
```
[the remark this answers is not in the document]
Comment (sidvishnoi) [reply src/core/dfn-panel.js]: Still need to get `firstElementChild` at least. Can't style a `DocumentFragment` 😅

**Comment by sidvishnoi** [comment]
Closed in favor of https://github.com/w3c/respec/pull/2820

**Pull request closed without merging**